tests/test_utils.py

```diff
@@ -35,3 +58,43 @@ def test_memory_human_readable_conversion():
     assert bytes_to_human_readable(1024) == (1.0, "KB")
     assert bytes_to_human_readable(1024**2) == (1.0, "MB")
     assert bytes_to_human_readable(4.3 * 1024**4) == (4.3, "TB")
+
+
+@pytest.mark.parametrize(
+    ("target, hook, error"),
+    [
+        param(base_func, func_with_default_values, None, id="hook_with_default_values"),
+        param(
+            func_with_default_values, base_func, None, id="target_with_default_values"
+        ),
+        param(base_func, func_without_annotations, None, id="hook_has_no_annotations"),
+        param(
+            base_func,
+            func_with_different_order,
+            TypeError,
+            id="hook_has_different_signature_order",
+        ),
+        param(
+            func_with_different_order,
+            base_func,
+            TypeError,
+            id="target_has_different_signature_order",
+        ),
+        param(
+            base_func,
+            func_with_different_names,
+            TypeError,
+            id="hook_has_different_names",
+        ),
+        param(
+            func_with_different_order,
```

Comment:
is this a duplicate of the 3rd last entry?
prob `func_with_different_names` is meant here?

Reply:
Oh true, I fixed it 👍